src/Microsoft.Performance.SDK/Processing/CodeGenerationContext.cs

```diff
@@ -110,7 +110,7 @@ private static Type CreateStructFunctionType(MethodInfo methodToCall)
             var resultType = methodToCall.ReturnType;
 
             var type = moduleBuilder.DefineType(
-                methodToCall.DeclaringType.FullName + "$." + methodToCall.Name,
+                $"{methodToCall.DeclaringType.FullName}$.{methodToCall.Name}$.{methodToCall.MethodHandle.Value.ToInt64():x16}",
```

Comment:
Thanks for this Cristian!

Would you mind adding a comment here to explain that we're using the MethodHandle as a way to uniquely define the type name in cases where there might be a name collision.

It is obvious in context of this PR, but some poor dev looking at this in 3 years might be scratching their head.

Reply:
Given that plugin assemblies may be loaded and unloaded within a given process, and that the result of this method is stored in a static value which may remain loaded, if using the method handle is safe.

Reply:
I think this will be ok because it's keyed on the MethodInfo and so the name shouldn't be reused in this case.

Reply:
Added comment explaining intent and effectiveness of using `MethodInfo.MethodHandle.Value` in the name of the generated type, including why the `methodToCall` cannot be unloaded while the generated type is still loaded.